src/m365/teams/commands/meeting/meeting-attendacereport-list.spec.ts

```diff
@@ -0,0 +1,229 @@
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(false);
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/me/onlineMeetings/${meetingId}/attendanceReports`) {
+        return { value: meetingAttendanceResponse };
+      }
+      throw 'Invalid request.';
+    });
+
+    await command.action(logger, {
+      options:
+      {
+        meetingId: meetingId
+      }
+    });
+
+    assert(loggerLogSpy.calledWith(meetingAttendanceResponse));
+  });
+
+  it('retrieves meeting attendace reports correctly by userId', async () => {
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userId}/onlineMeetings/${meetingId}/attendanceReports`) {
+        return { value: meetingAttendanceResponse };
+      }
+      throw 'Invalid request.';
+    });
+
+    await command.action(logger, {
+      options:
+      {
+        meetingId: meetingId,
+        userId: userId
+      }
+    });
+
+    assert(loggerLogSpy.calledWith(meetingAttendanceResponse));
+  });
+
+  it('retrieves meeting attendace reports correctly by userName', async () => {
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userName}/onlineMeetings/${meetingId}/attendanceReports`) {
+        return { value: meetingAttendanceResponse };
+      }
+      throw 'Invalid request.';
+    });
+
+    await command.action(logger, {
+      options:
+      {
+        meetingId: meetingId,
+        userName: userName
+      }
+    });
+
+    assert(loggerLogSpy.calledWith(meetingAttendanceResponse));
+  });
+
+  it('retrieves meeting attendace reports correctly by userEmail', async () => {
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    sinon.stub(Cli, 'executeCommandWithOutput').callsFake(async (command): Promise<any> => {
+      if (command === userGetCommand) {
+        return { "stdout": JSON.stringify({ id: userId }) };
+      }
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userId}/onlineMeetings/${meetingId}/attendanceReports`) {
+        return { value: meetingAttendanceResponse };
+      }
+      throw 'Invalid request.';
+    });
+
+    await command.action(logger, {
+      options:
+      {
+        meetingId: meetingId,
+        email: userName,
+        verbose: true
+      }
+    });
+
+    assert(loggerLogSpy.calledWith(meetingAttendanceResponse));
+  });
+
+  it('correctly handles error when throwing request', async () => {
+    sinon.stub(request, 'get').callsFake(async () => {
+      throw 'An error has occurred';
```

Comment:
It's very unlikely that the API will actually throw a plain error message. Instead we should use a more realistic error message.